code_generation/templates/power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_gen/attribute_classes.hpp.jinja

```diff
@@ -29,14 +30,12 @@ struct get_attributes_list<{{ attribute_class.name }}<sym>> {
 template<>
 struct get_attributes_list<{{ attribute_class.name }}> {
 {%- endif %}
-    std::vector<MetaAttribute> operator() () const {
-        // all attributes including base class
-        return {
+    static constexpr std::array<MetaAttribute, {{ attribute_class.full_attributes|length }}> value{
```

Comment:
the compiler is able to deduce the length of the array and it will be much more beautiful

```suggestion
    static constexpr std::array value{
```

if that doesn't work, then you'll have to do something like

```cpp
    static constexpr std::array value{
            MetaAttribute{meta_data_gen::get_meta_attribute<{{ attribute_class.full_name }}, &{{ attribute_class.full_name }}::{{ attribute.names }}, offsetof({{ attribute_class.full_name }}, {{ attribute.names }}), []{ return "{{ attribute.names }}"; }>::value},
```

Reply:
Compiler must deduce type and size at the same time. It is possible here. But we are using code generation. So to generate it explicitly should not hurt.